### ISSUE TITLE
Step is now broken when calling pip

### ISSUE THREAD

**Issue by that-guy-iain** [unfor19/install-aws-cli-action#31]
Returns an error

```
Running cmd: /usr/local}/aws/bin/python -m pip install --no-binary :all: --no-cache-dir --no-index --find-links file://. setuptools_scm-3.3.3.tar.gz
Traceback (most recent call last):
  File "/home/runner/work/docs/docs/unfor19-awscli/./awscli-bundle/install", line 258, in <module>
    main()
  File "/home/runner/work/docs/docs/unfor19-awscli/./awscli-bundle/install", line 238, in main
    pip_install_packages(opts.install_dir)
  File "/home/runner/work/docs/docs/unfor19-awscli/./awscli-bundle/install", line 148, in pip_install_packages
    _install_setup_deps(python, '.')
  File "/home/runner/work/docs/docs/unfor19-awscli/./awscli-bundle/install", line 167, in _install_setup_deps
    run(
  File "/home/runner/work/docs/docs/unfor19-awscli/./awscli-bundle/install", line 88, in run
    raise BadRCError("Bad rc (%s) for cmd '%s': %s" % (
BadRCError: Bad rc (1) for cmd '/usr/local}/aws/bin/python -m pip install --no-binary :all: --no-cache-dir --no-index --find-links file://. setuptools_scm-3.3.3.tar.gz': Looking in links: file://.
Processing ./setuptools_scm-3.3.3.tar.gz
  Installing build dependencies: started
  Installing build dependencies: finished with status 'error'
  error: subprocess-exited-with-error
  
  × pip subprocess to install build dependencies did not run successfully.
  │ exit code: 2
  ╰─> [55 lines of output]
      WARNING: The directory '/home/runner/.cache/pip' or its parent directory is not owned or is not writable by the current user. The cache has been disabled. Check the permissions and owner of that directory. If executing pip with sudo, you should use sudo's -H flag.
      Looking in links: file://.
      ERROR: Exception:
      Traceback (most recent call last):
        File "/usr/local}/aws/lib/python3.12/site-packages/pip/_internal/cli/base_command.py", line 180, in exc_logging_wrapper
          status = run_func(*args)
                   ^^^^^^^^^^^^^^^
        File "/usr/local}/aws/lib/python3.12/site-packages/pip/_internal/cli/req_command.py", line 245, in wrapper
          return func(self, options, args)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^
        File "/usr/local}/aws/lib/python3.12/site-packages/pip/_internal/commands/install.py", line 377, in run
          requirement_set = resolver.resolve(
                            ^^^^^^^^^^^^^^^^^
        File "/usr/local}/aws/lib/python3.12/site-packages/pip/_internal/resolution/resolvelib/resolver.py", line 95, in resolve
          result = self._result = resolver.resolve(
                                  ^^^^^^^^^^^^^^^^^
        File "/usr/local}/aws/lib/python3.12/site-packages/pip/_vendor/resolvelib/resolvers.py", line [54](https://github.com/billabear/docs/actions/runs/11307682142/job/31449759324#step:6:57)6, in resolve
          state = resolution.resolve(requirements, max_rounds=max_rounds)
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "/usr/local}/aws/lib/python3.12/site-packages/pip/_vendor/resolvelib/resolvers.py", line 397, in resolve
          self._add_to_criteria(self.state.criteria, r, parent=None)
        File "/usr/local}/aws/lib/python3.12/site-packages/pip/_vendor/resolvelib/resolvers.py", line 173, in _add_to_criteria
          if not criterion.candidates:
                 ^^^^^^^^^^^^^^^^^^^^
        File "/usr/local}/aws/lib/python3.12/site-packages/pip/_vendor/resolvelib/structs.py", line 156, in __bool__
          return bool(self._sequence)
                 ^^^^^^^^^^^^^^^^^^^^
        File "/usr/local}/aws/lib/python3.12/site-packages/pip/_internal/resolution/resolvelib/found_candidates.py", line 1[55](https://github.com/billabear/docs/actions/runs/11307682142/job/31449759324#step:6:58), in __bool__
          return any(self)
                 ^^^^^^^^^
        File "/usr/local}/aws/lib/python3.12/site-packages/pip/_internal/resolution/resolvelib/found_candidates.py", line 143, in <genexpr>
          return (c for c in iterator if id(c) not in self._incompatible_ids)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "/usr/local}/aws/lib/python3.12/site-packages/pip/_internal/resolution/resolvelib/found_candidates.py", line 44, in _iter_built
          for version, func in infos:
        File "/usr/local}/aws/lib/python3.12/site-packages/pip/_internal/resolution/resolvelib/factory.py", line 297, in iter_index_candidate_infos
          result = self._finder.find_best_candidate(
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "/usr/local}/aws/lib/python3.12/site-packages/pip/_internal/index/package_finder.py", line 890, in find_best_candidate
          candidates = self.find_all_candidates(project_name)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "/usr/local}/aws/lib/python3.12/site-packages/pip/_internal/index/package_finder.py", line 817, in find_all_candidates
          collected_sources = self._link_collector.collect_sources(
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "/usr/local}/aws/lib/python3.12/site-packages/pip/_internal/index/collector.py", line 480, in collect_sources
          find_links_sources = collections.OrderedDict(
                               ^^^^^^^^^^^^^^^^^^^^^^^^
        File "/usr/local}/aws/lib/python3.12/site-packages/pip/_internal/index/collector.py", line 4[81](https://github.com/billabear/docs/actions/runs/11307682142/job/31449759324#step:6:84), in <genexpr>
          build_source(
        File "/usr/local}/aws/lib/python3.12/site-packages/pip/_internal/index/sources.py", line 242, in build_source
          path = url_to_path(location)
                 ^^^^^^^^^^^^^^^^^^^^^
        File "/usr/local}/aws/lib/python3.12/site-packages/pip/_internal/utils/urls.py", line 43, in url_to_path
          raise ValueError(
      ValueError: non-local file URIs are not supported on this platform: 'file://.'
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: subprocess-exited-with-error

× pip subprocess to install build dependencies did not run successfully.
│ exit code: 2
╰─> See above for output.

note: This error originates from a subprocess, and is likely not a problem with pip.
```

**Comment by unfor19** [comment]
@that-guy-iain Thank you for your input

I've executed a test run - https://github.com/unfor19/install-aws-cli-action/actions/runs/11307805272

Everything seems to be working, can you please provide more context?

**Comment by that-guy-iain** [comment]
https://github.com/billabear/docs/actions/runs/11307682142/job/31449759324 is the broken build

**Comment by unfor19** [comment]
@that-guy-iain thanks!

From first glance, comparing to previous builds, you were running on **ubuntu-22.04** while your latest build which fails runs on **ubuntu-24.04**.

The upgrade of OS happened because you’re using “runs-on: ubuntu-latest” and I guess Github decided this weekend that “latest” means 24.04 rather than 22.04.

For a quick fix - set the “runs-on: ubuntu-22.04” and I assume you’ll be fine, feel free to report back.

I will update this action, aws-cli-action, tomorrow  to support ubuntu-24.04 if it’s possible.

And again, thank you for your input 🙏🏻

**Comment by that-guy-iain** [comment]
Yea, that change worked. Thanks!
